src/IIIFPresentation/API/Converters/CollectionConverter.cs

```diff
@@ -1,20 +1,18 @@
-﻿using API.Infrastructure.Helpers;
+﻿using Core.Helpers;
 using IIIF.Presentation.V3;
-using IIIF.Presentation.V3.Annotation;
-using IIIF.Presentation.V3.Content;
 using Models.API.Collection;
 using Models.Infrastucture;
 
 namespace API.Converters;
 
 public static class CollectionConverter
 {
-    public static Collection ToHierarchicalCollection(this Models.Database.Collections.Collection dbAsset, UrlRoots urlRoots,
-        IQueryable<Models.Database.Collections.Collection>? items)
+    public static Collection ToHierarchicalCollection(this Models.Database.Collections.Collection dbAsset, 
+        UrlRoots urlRoots, List<Models.Database.Collections.Collection>? items)
     {
-        var stuff = new Collection()
+        var collection = new Collection()
         {
-            Id = $"{urlRoots.BaseUrl}/{dbAsset.CustomerId}{(dbAsset.FullPath != null ? $"/{dbAsset.FullPath}" : "")}",
+            Id = $"{urlRoots.BaseUrl}/{dbAsset.CustomerId}{(string.IsNullOrEmpty(dbAsset.FullPath) ? string.Empty : $"/{dbAsset.FullPath}")}",
             Context = "http://iiif.io/api/presentation/3/context.json",
```

Comment:
As this is a `Collection` from the iiif-net package there's a couple of helpers for adding contexts. It ensures that they're ordered correctly and avoids adding multiple times etc.

```cs
collection.EnsurePresentation3Context();
// which is a shortcut for...
collection.EnsureContext(IIIF.Presentation.Context.Presentation3Context);
```